pinakes/main/catalog/services/copy_portfolio_item.py

```diff
@@ -25,15 +19,19 @@
 class CopyPortfolioItem:
     """Copy portfolio item service"""
 
-    def __init__(self, portfolio_item, options):
+    def __init__(
+        self,
+        portfolio_item: PortfolioItem,
+        portfolio: Optional[Portfolio] = None,
+        portfolio_item_name: Optional[str] = None,
+    ):
         self.portfolio_item = portfolio_item
-        self.name = options.get("portfolio_item_name", portfolio_item.name)
-
-        portfolio_id = options.get("portfolio_id", None)
-        if portfolio_id:
-            self.portfolio = Portfolio.objects.get(id=portfolio_id)
-        else:
-            self.portfolio = self.portfolio_item.portfolio
+        if portfolio is None:
+            portfolio = self.portfolio_item.portfolio
+        self.portfolio = portfolio
+        if portfolio_item_name is None:
+            portfolio_item_name = self.portfolio_item.name
```

Comment:
Wait a second I think I'm confused, how does this not duplicate the name, it must be handled somewhere later, right? Otherwise we would need to use the `_new_portfolio_item_name` method to make sure it's not a duplicate name?

I ask because it looks like the tests seem to still work with https://github.com/ansible/pinakes/pull/515/files#diff-498a816482356965fb885795c457b89e43f9a2b7410e5d8a89b47aa1ba8ba969R107 which I'm pretty sure uses the same name, so either I'm missing something or the test isn't testing what I think it is.

Reply:
The `_new_portfolio_item_name` is called unconditionally when a new name is generated

https://github.com/ansible/pinakes/pull/515/files#diff-ca0da050bd26f2528aa2ed7af4ec8010fb22eaf728359e9470343e559adcbbc7L62